docs/types.md

```diff
@@ -783,6 +784,9 @@ All types of FileSpecs share some common fields:
 ### FileSpec -- Path
 
 A FileSpec with `type` equal to `path` points to an absolute or relative path accessible from the current machine.
+The path may also include a glob pattern to target multiple files or directories.
+For glob patterns resolving to multiple files, files will be copied into the specified `dest`.
```

Comment:
This description needs work.
I think you made it sound too fancy and missed a key point about dest (that it will be treated as a dir).
You need more about dest being treated as a dir, but start simple like "For paths that match multiple dirents, ...".